tests/bin/difference.py

```diff
@@ -0,0 +1,406 @@
+         'help': "Command to invoke litani.  Defaults to '%(default)s'."},
+        {'flag': '--force',
+         'action': 'store_true',
+         'help': 'Overwrite existing report or installation directories'},
+    ]
+    args = arguments.create_parser(options, description).parse_args()
+    arguments.configure_logging(args)
+    return args
+
+################################################################
+# Run a command
+
+def run(cmd, stdin=None, check=True, env=None, cwd=None, capture_output=True):
+    """Run a command as a subprocess"""
+
+    cmd = [str(word) for word in cmd]
+    stdin = str(stdin) if stdin is not None else None
+    kwds = {
+        "env": env,
+        "cwd": cwd,
+        "text": True,
+        "stdout": subprocess.PIPE if capture_output else None,
+        "stderr": subprocess.PIPE if capture_output else None,
+    }
+    with subprocess.Popen(cmd, **kwds) as pipe:
+        stdout, stderr = pipe.communicate(input=stdin)
+    if check and pipe.returncode:
+        logging.debug("Command %s returned %s", cmd, pipe.returncode)
+        for line in stderr.splitlines():
+            logging.debug(line)
+        raise UserWarning(f"Command '{' '.join(cmd)}' returned '{pipe.returncode}'")
+    if not capture_output:
+        return None
+    return [line.rstrip() for line in stdout.splitlines()]
+
+################################################################
+
+def install_viewer(viewer_commit, venv_root, viewer_repo):
+    """Install a given version of viewer into a virtual environment"""
+
+    # save the original commit
+    branch = run(["git", "branch", "--show-current"], cwd=viewer_repo)[0]
+    run(["git", "stash"], cwd=viewer_repo)
+
+    try:
+        logging.info("Setting up virtual environment %s for viewer %s", venv_root, viewer_commit)
+        run(["python3", "-m", "venv", venv_root])
+        venv_bin = Path(venv_root) / "bin"
+        env = dict(os.environ.items())
+        env['PATH'] = str(venv_bin) + os.pathsep + env['PATH']
+        run(["python3", "-m", "pip", "install", "--upgrade", "pip", "setuptools", "wheel", "build"],
+            env=env)
+
+        logging.info("Installing viewer %s into virtual environment %s", viewer_commit, venv_root)
+        run(["git", "checkout", viewer_commit], cwd=viewer_repo, env=env)
+        run(["make", "install"], cwd=viewer_repo, env=env)
+
+        venv_viewer = venv_bin / "cbmc-viewer"
+        return venv_viewer
+    except UserWarning as error:
+        logging.info("Failed to install viewer %s into virtual environment %s",
+                     viewer_commit, venv_root)
+        logging.info(error)
+        raise
+
+    finally:
+        logging.info("Restoring viewer repository to %s", branch)
+        run(["git", "checkout", branch], cwd=viewer_repo)
+        try:
+            run(["git", "stash", "pop"], cwd=viewer_repo)
+        except UserWarning:
+            # Probably means prior 'git stash' found nothing to stash
```

Comment:
It's fine to leave this for a future PR, I just wanted to point out that you already implemented a `check_git_status` method below that does this